superduperdb/components/model.py

```diff
@@ -509,6 +509,18 @@ def __post_init__(self, db, artifacts):
         if not self.identifier:
             raise Exception('_Predictor identifier must be non-empty')
 
+    def pre_create(self, db):
+        """Method to run before creating the model in the database.
+
+        :param db: Datalayer instance.
+        """
+        if isinstance(self.datatype, DataType):
+            from superduperdb.misc.datatype_helper import convert_model_datatype
+
+            is_ibis = db.databackend.db_type == DBType.SQL
+            self.datatype = convert_model_datatype(self.datatype, is_ibis)
+        super().pre_create(db)
+
```

Comment:
I don't see what this logic is for.

Reply:
To adapt data types for different data backends, such as vector to sqlvector, users don’t need to worry about whether they are using vector or sqlvector and can simply use vector.

This is not a bug fix and can be removed. After removing it, explicit use of vector/sqlvector for MongoDB/ibis is required.

WDYT?

Reply:
Removed